treescript/src/treescript/task.py

```diff
@@ -193,39 +186,35 @@ def get_ignore_closed_tree(task):
 
 # task_action_types {{{1
 def task_action_types(config, task):
-    """Extract task actions as scope definitions.
+    """Extract task actions from task payload.
 
     Args:
         config (dict): the running config.
         task (dict): the task definition.
 
     Raises:
-        TaskVerificationError: if the number of cert scopes is not 1.
+        TaskVerificationError: if unknown actions are specified.
 
     Returns:
-        str: the cert type.
+        set: the set of specified actions
 
     """
-    if task.get("payload", {}).get("actions"):
-        actions = task["payload"]["actions"]
-    else:
-        log.warning("Scopes-based actions are deprecated! Use task.payload.actions instead.")
-        actions = [s.split(":")[-1] for s in task["scopes"] if s.startswith(config["taskcluster_scope_prefix"] + "action:")]
-        if len(actions) < 1:
-            raise TaskVerificationError("Need at least one valid action specified in scopes")
+    actions = set(task["payload"].get("actions", []))
     log.info("Action requests: %s", actions)
-    invalid_actions = set(actions) - set(VALID_ACTIONS)
+    invalid_actions = actions - VALID_ACTIONS
     if len(invalid_actions) > 0:
         raise TaskVerificationError("Task specified invalid actions: {}".format(invalid_actions))
```

Comment:
It looks like we no longer have coverage for this line; could you please add it?